lib/src/entrypoint.dart

```diff
@@ -840,6 +845,25 @@ class Entrypoint {
     }
   }
 
+  /// Checks whether or not the `.dart_tool/package_config.json` file is was
+  /// generated by a different sdk down changes in minor versions.
+  ///
+  /// For pre-releases we always consider the package_config.json out of date
+  /// when the version changes.
+  ///
+  /// Throws [DataException], if `.dart_tool/package_config.json` the version
+  /// changed sufficiently.
+  void _checkPackageConfigSameSdk() {
+    final generatorVersion = packageConfig.generatorVersion;
+    if (generatorVersion == null ||
+        generatorVersion.major != sdk.version.major ||
+        generatorVersion.minor != sdk.version.minor ||
+        generatorVersion.isPreRelease ||
+        sdk.version.isPreRelease) {
+      dataError('The sdk was updated since last package resolution');
```

Comment:
Should we change to ensureUpToDate...
And avoid any risk that these messages become user visible

Reply:
Probably a future refactor...

Reply:
Yeah - I think we can do that when we provide the functionality from the external API.